pkg/plugins/gateway/appservice/http.go

```diff
@@ -61,11 +62,23 @@ func handleNotifications(ctx *fasthttp.RequestCtx, events []eventgrid.Event, poo
 			payloadBytes, _ = json.Marshal(event.Data)
 		}
 
-		evt := &triggers.Event{
-			// FIXME: Check if ID is nil
-			ID:      *event.ID,
-			Topic:   *event.Topic,
-			Payload: payloadBytes,
+		nitricEvt := &ep.NitricEvent{}
+		var evt *triggers.Event
+		// attempt to deserialize as a nitric event
+		if err := json.Unmarshal(payloadBytes, nitricEvt); err == nil && nitricEvt.ID != "" {
+			payload, _ := json.Marshal(nitricEvt.Payload)
```

Comment:
check this error too?

Reply:
This diff looks out of date? I'm sure I've removed this code in handle notifications and unified how this is treated.

Reply:
let me check, I was reviewing commit by commit because it was a big PR